compiler/runtime/runtime.c

```diff
@@ -17,16 +17,24 @@
    tests this bit to determine if an object is a string or an object of some
    other type. */
 
+sk_string_t* get_sk_string(char* obj) {
+  return (sk_string_t*)(obj - sizeof(uint32_t) * 2);
```

Comment:
Should it be written
`(sk_string_t*)(obj - offsetof(sk_string_t, data))`?

Reply:
I thought about that but held off for now since it would take a bit more work to be able to do it consistently that way for all of strings, arrays, and general skip objects. I was also thinking that the compiler code produces values with exact knowledge about the layout, and writing the runtime code using `offsetof` gives the impression that things are robust with respect to the compiler making different decisions about struct layout such as adding padding, when in fact that is not true. But the runtime code is much easier to read using actual C types and just have in mind that we are still sensitive to the exact layout the compiler currently uses. So after your other changes, it would make sense to change this to use `offsetof` as well.